In [1]:
'''from IPython.core.display import display, HTML
display(HTML("<style>.container { width:90% !important; }</style>"))'''

'from IPython.core.display import display, HTML\ndisplay(HTML("<style>.container { width:90% !important; }</style>"))'

In [2]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import os


pd.options.display.max_columns = 999
pd.options.display.max_rows = 999

In [3]:
#return the first 5 and last 5 rows of this dataframe
def p(df_):
    if df_.shape[0] > 6:
        print(df_.shape)
        return pd.concat([df_.head(), df_.tail()])
    else:
        return df_

def rcp(file_, pd_=None):
    if pd_ == None:
        return pd.read_csv(os.path.join('..', 'processed_data', file_))
    else:
        return pd.read_csv(os.path.join('..', 'processed_data', file_), parse_dates=pd_)
    
def rcr(file_, pd_=None):
    if pd_ == None:
        return pd.read_csv(os.path.join('..', 'raw_data', file_))
    else:
        return pd.read_csv(os.path.join('..', 'raw_data', file_), parse_dates=pd_)

In [4]:
os.getcwd()

'/Users/pauljacob/Documents/GitHub/DSCapstone2/02_Data_Wrangling'

In [5]:
os.listdir()

['csvoutput_02_data_wrangling_6injuries_2010_11_player_replace_v1-Copy1.ipynb',
 'csvoutput_02_data_wrangling_0team_id_abbreviation_df.ipynb',
 'Untitled1.ipynb',
 'csvoutput_02_data_wrangling_6injuries_2011_12_date_range_check.ipynb',
 '.DS_Store',
 'csvoutput_02_data_wrangling_13player_id_player.ipynb',
 '02_data_wrangling_game_id_player_name_advanced_stats_api_calls_100_cuts.ipynb',
 'csvoutput_02_data_wrangling_6injuries_2010_11_player_replace_v2.ipynb',
 'Untitled3.ipynb',
 'csvoutput_02_data_wrangling_19boxscoreadvancedv2_team_2010_2018.ipynb',
 'xxx02_data_wrangling_nba_api_player_inactive_games_check.ipynb',
 'Untitled.ipynb',
 'Untitled4.ipynb',
 'csvoutput_02_data_wrangling_3game_id_game_date_home_away_V2.ipynb',
 'csvoutput_02_data_wrangling_10team_id_season_team.ipynb',
 'csvoutput_02_data_wrangling_18game_id_team_id_home_away.ipynb',
 'csvoutput_02_data_wrangling_4game_id_team_stats_betting_totals_spreads_money_line.ipynb',
 '02_data_wrangling_contribution_lost_injury_inac

In [6]:
listtt = os.listdir(os.path.join('..', 'processed_data_preliminary'))
listtt.remove('.DS_Store')
listtt

['5game_id_team_id_advanced_stats5880.csv',
 '5game_id_team_id_advanced_stats6345.csv',
 '5game_id_team_id_advanced_stats8340.csv',
 '5game_id_team_id_advanced_stats5670.csv',
 '5game_id_team_id_player_name_advanced_stats3175.csv',
 '5game_id_team_id_advanced_stats14805.csv',
 '5game_id_team_id_advanced_stats11955.csv',
 '5game_id_team_id_advanced_stats13830.csv',
 '5game_id_team_id_advanced_stats12045.csv',
 '5game_id_team_id_player_name_advanced_stats13831.csv',
 '5game_id_team_id_advanced_stats3765.csv',
 '5game_id_team_id_advanced_stats10875.csv',
 '5game_id_team_id_advanced_stats2490.csv',
 '5game_id_team_id_advanced_stats14385.csv',
 '5game_id_team_id_advanced_stats405.csv',
 '5game_id_team_id_advanced_stats13575.csv',
 '5game_id_team_id_player_name_advanced_stats13979.csv',
 '5game_id_team_id_advanced_stats9300.csv',
 '5game_id_team_id_advanced_stats7305.csv',
 '5game_id_team_id_player_name_advanced_stats1399.csv',
 '5game_id_team_id_advanced_stats12870.csv',
 '5game_id_team_id_

In [7]:
listttt = [k for k in listtt if 'player_name' not in k]
listttt

['5game_id_team_id_advanced_stats5880.csv',
 '5game_id_team_id_advanced_stats6345.csv',
 '5game_id_team_id_advanced_stats8340.csv',
 '5game_id_team_id_advanced_stats5670.csv',
 '5game_id_team_id_advanced_stats14805.csv',
 '5game_id_team_id_advanced_stats11955.csv',
 '5game_id_team_id_advanced_stats13830.csv',
 '5game_id_team_id_advanced_stats12045.csv',
 '5game_id_team_id_advanced_stats3765.csv',
 '5game_id_team_id_advanced_stats10875.csv',
 '5game_id_team_id_advanced_stats2490.csv',
 '5game_id_team_id_advanced_stats14385.csv',
 '5game_id_team_id_advanced_stats405.csv',
 '5game_id_team_id_advanced_stats13575.csv',
 '5game_id_team_id_advanced_stats9300.csv',
 '5game_id_team_id_advanced_stats7305.csv',
 '5game_id_team_id_advanced_stats12870.csv',
 '5game_id_team_id_advanced_stats11160.csv',
 '5game_id_team_id_advanced_stats6225.csv',
 '5game_id_team_id_advanced_stats13005.csv',
 '5game_id_team_id_advanced_stats10530.csv',
 '5game_id_team_id_advanced_stats3375.csv',
 '5game_id_team_id_adv

In [8]:
len(listttt)

1035

In [9]:
listttt[0]

'5game_id_team_id_advanced_stats5880.csv'

In [10]:
df_empty = pd.DataFrame({})
df_empty

""


In [11]:
for i in range(len(listttt)):
    print(i)
    df_temp = pd.read_csv(os.path.join('..', 'processed_data_preliminary', str(listttt[i])))
    df_empty = pd.concat([df_empty, df_temp])

0
1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89
90
91
92
93
94
95
96
97
98
99
100
101
102
103
104
105
106
107
108
109
110
111
112
113
114
115
116
117
118
119
120
121
122
123
124
125
126
127
128
129
130
131
132
133
134
135
136
137
138
139
140
141
142
143
144
145
146
147
148
149
150
151
152
153
154
155
156
157
158
159
160
161
162
163
164
165
166
167
168
169
170
171
172
173
174
175
176
177
178
179
180
181
182
183
184
185
186
187
188
189
190
191
192
193
194
195
196
197
198
199
200
201
202
203
204
205
206
207
208
209
210
211
212
213
214
215
216
217
218
219
220
221
222
223
224
225
226
227
228
229
230
231
232
233
234
235
236
237
238
239
240
241
242
243
244
245
246
247
248
249
250
251
252
253
254
255
256
257
258
259
260
261
262
263
264
265
266
267
268
269
270
271
272
273
274
275
276
27

In [12]:
df_empty.head()

,GAME_ID,TEAM_ID,TEAM_NAME,TEAM_ABBREVIATION,TEAM_CITY,MIN,E_OFF_RATING,OFF_RATING,E_DEF_RATING,DEF_RATING,E_NET_RATING,NET_RATING,AST_PCT,AST_TOV,AST_RATIO,OREB_PCT,DREB_PCT,REB_PCT,E_TM_TOV_PCT,TM_TOV_PCT,EFG_PCT,TS_PCT,USG_PCT,E_USG_PCT,E_PACE,PACE,PACE_PER40,POSS,PIE
0,21000961,1610612740,Hornets,NOH,New Orleans,240:00,102.5,106.9,102.1,105.7,0.3,1.1,0.382,1.44,11.5,0.319,NaN,NaN,9.916,10.3,0.462,0.512,1.0,0.198,90.42,87.0,72.50,87,0.479
1,21000961,1610612742,Mavericks,DAL,Dallas,240:00,102.1,105.7,102.5,106.9,-0.3,-1.1,0.710,2.00,19.0,NaN,0.681,NaN,12.211,12.6,0.493,0.554,1.0,0.191,90.42,87.0,72.50,87,0.521
2,21000962,1610612759,Spurs,SAS,San Antonio,240:00,125.0,124.7,110.9,118.2,14.1,6.5,0.467,1.40,18.3,0.222,0.556,0.444,16.892,16.9,0.686,0.704,1.0,0.198,91.30,88.5,73.75,89,0.610
3,21000962,1610612765,Pistons,DET,Detroit,240:00,110.9,118.2,125.0,124.7,-14.1,-6.5,0.524,1.83,16.2,0.444,0.778,0.556,12.793,13.6,0.484,0.511,1.0,0.197,91.30,88.5,73.75,88,0.390
4,21000963,1610612758,Kings,SAC,Sacramento,240:00,97.0,101.0,104.9,106.0,-7.9,-5.0,0.529,0.86,12.9,0.370,0.841,0.582,19.970,20.8,0.440,0.509,1.0,0.196,103.12,100.5,83.75,101,0.478


In [13]:
df_empty.reset_index(drop=True, inplace=True)
p(df_empty)

(31036, 29)


,GAME_ID,TEAM_ID,TEAM_NAME,TEAM_ABBREVIATION,TEAM_CITY,MIN,E_OFF_RATING,OFF_RATING,E_DEF_RATING,DEF_RATING,E_NET_RATING,NET_RATING,AST_PCT,AST_TOV,AST_RATIO,OREB_PCT,DREB_PCT,REB_PCT,E_TM_TOV_PCT,TM_TOV_PCT,EFG_PCT,TS_PCT,USG_PCT,E_USG_PCT,E_PACE,PACE,PACE_PER40,POSS,PIE
0,21000961,1610612740,Hornets,NOH,New Orleans,240:00,102.5,106.9,102.1,105.7,0.3,1.1,0.382,1.44,11.5,0.319,NaN,NaN,9.916,10.3,0.462,0.512,1.0,0.198,90.42,87.0,72.50,87,0.479
1,21000961,1610612742,Mavericks,DAL,Dallas,240:00,102.1,105.7,102.5,106.9,-0.3,-1.1,0.710,2.00,19.0,NaN,0.681,NaN,12.211,12.6,0.493,0.554,1.0,0.191,90.42,87.0,72.50,87,0.521
2,21000962,1610612759,Spurs,SAS,San Antonio,240:00,125.0,124.7,110.9,118.2,14.1,6.5,0.467,1.40,18.3,0.222,0.556,0.444,16.892,16.9,0.686,0.704,1.0,0.198,91.30,88.5,73.75,89,0.610
3,21000962,1610612765,Pistons,DET,Detroit,240:00,110.9,118.2,125.0,124.7,-14.1,-6.5,0.524,1.83,16.2,0.444,0.778,0.556,12.793,13.6,0.484,0.511,1.0,0.197,91.30,88.5,73.75,88,0.390
4,21000963,1610612758,Kings,SAC,Sacramento,240:00,97.0,101.0,104.9,106.0,-7.9,-5.0,0.529,0.86,12.9,0.370,0.841,0.582,19.970,20.8,0.440,0.509,1.0,0.196,103.12,100.5,83.75,101,0.478
31031,21700898,1610612738,Celtics,BOS,Boston,240:00,112.5,111.2,100.9,101.0,11.6,10.2,0.641,3.13,19.1,0.231,0.830,0.515,8.258,8.2,0.523,0.557,1.0,0.200,96.98,97.5,81.25,98,0.529
31032,21700899,1610612741,Bulls,CHI,Chicago,240:00,85.2,87.0,102.4,104.0,-17.2,-17.0,0.515,1.21,12.7,0.297,0.840,0.535,13.709,14.0,0.387,0.422,1.0,0.198,101.82,100.0,83.33,100,0.408
31033,21700899,1610612751,Nets,BKN,Brooklyn,240:00,102.4,104.0,85.2,87.0,17.2,17.0,0.775,2.07,22.2,0.160,0.703,0.465,14.775,15.0,0.533,0.556,1.0,0.195,101.82,100.0,83.33,100,0.592
31034,21700900,1610612752,Knicks,NYK,New York,240:00,106.8,109.9,124.5,121.4,-17.7,-11.5,0.596,2.00,19.7,0.265,0.762,0.495,13.467,13.9,0.542,0.555,1.0,0.200,102.20,102.0,85.00,101,0.421
31035,21700900,1610612744,Warriors,GSW,Golden State,240:00,124.5,121.4,106.8,109.9,17.7,11.5,0.705,2.58,22.2,0.238,0.735,0.505,11.947,11.7,0.606,0.648,1.0,0.201,102.20,102.0,85.00,103,0.579


In [14]:
31036 / 2

15518.0

<font color='red'> this number of games correct for seasons 2006-07 thru 2017-18???</font> -yes, 15519 - 1 = 15518 for a post boston marathon game not played between the pacers and celtics

In [15]:
df_empty.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 31036 entries, 0 to 31035
Data columns (total 29 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   GAME_ID            31036 non-null  int64  
 1   TEAM_ID            31036 non-null  int64  
 2   TEAM_NAME          31036 non-null  object 
 3   TEAM_ABBREVIATION  31036 non-null  object 
 4   TEAM_CITY          31036 non-null  object 
 5   MIN                31036 non-null  object 
 6   E_OFF_RATING       31036 non-null  float64
 7   OFF_RATING         31036 non-null  float64
 8   E_DEF_RATING       31036 non-null  float64
 9   DEF_RATING         31036 non-null  float64
 10  E_NET_RATING       31036 non-null  float64
 11  NET_RATING         31036 non-null  float64
 12  AST_PCT            31036 non-null  float64
 13  AST_TOV            31036 non-null  float64
 14  AST_RATIO          31036 non-null  float64
 15  OREB_PCT           26705 non-null  float64
 16  DREB_PCT           267

In [16]:
#df_empty.MIN.str.replace(':00', '').value_counts()

In [17]:
yoyo = df_empty.MIN.str.split(':')
type(yoyo)

pandas.core.series.Series

In [18]:
import numpy as np


In [19]:
df_ms = pd.DataFrame({})

In [20]:
df_ms[['minutes', 'seconds']] = pd.DataFrame(yoyo.tolist())
df_ms

,minutes,seconds
0,240,00
1,240,00
2,240,00
3,240,00
4,240,00
...,...,...
31031,240,00
31032,240,00
31033,240,00
31034,240,00


In [21]:
df_ms = df_ms.astype('int64')
p(df_ms)

(31036, 2)


,minutes,seconds
0,240,0
1,240,0
2,240,0
3,240,0
4,240,0
31031,240,0
31032,240,0
31033,240,0
31034,240,0
31035,240,0


In [22]:
df_ms[['hours']] = np.floor(df_ms.minutes / 60)
p(df_ms)

(31036, 3)


,minutes,seconds,hours
0,240,0,4.0
1,240,0,4.0
2,240,0,4.0
3,240,0,4.0
4,240,0,4.0
31031,240,0,4.0
31032,240,0,4.0
31033,240,0,4.0
31034,240,0,4.0
31035,240,0,4.0


In [23]:
df_ms.minutes = df_ms.minutes - (np.floor(df_ms.minutes / 60) * 60)
p(df_ms)

(31036, 3)


,minutes,seconds,hours
0,0.0,0,4.0
1,0.0,0,4.0
2,0.0,0,4.0
3,0.0,0,4.0
4,0.0,0,4.0
31031,0.0,0,4.0
31032,0.0,0,4.0
31033,0.0,0,4.0
31034,0.0,0,4.0
31035,0.0,0,4.0


In [24]:
df_ms.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 31036 entries, 0 to 31035
Data columns (total 3 columns):
 #   Column   Non-Null Count  Dtype  
---  ------   --------------  -----  
 0   minutes  31036 non-null  float64
 1   seconds  31036 non-null  int64  
 2   hours    31036 non-null  float64
dtypes: float64(2), int64(1)
memory usage: 727.5 KB


In [25]:
df_ms.value_counts('minutes')

minutes
0.0     29152
25.0     1590
50.0      232
15.0       48
59.0        8
40.0        6
dtype: int64

In [26]:
df_ms = df_ms.astype('int64')
p(df_ms)

(31036, 3)


,minutes,seconds,hours
0,0,0,4
1,0,0,4
2,0,0,4
3,0,0,4
4,0,0,4
31031,0,0,4
31032,0,0,4
31033,0,0,4
31034,0,0,4
31035,0,0,4


In [27]:
df_ms.value_counts('minutes')

minutes
0     29152
25     1590
50      232
15       48
59        8
40        6
dtype: int64

In [28]:
indexes_min_zero = df_ms.minutes[df_ms.minutes == 0].index
indexes_min_zero

Int64Index([    0,     1,     2,     3,     4,     5,     6,     7,     8,
                9,
            ...
            31026, 31027, 31028, 31029, 31030, 31031, 31032, 31033, 31034,
            31035],
           dtype='int64', length=29152)

In [29]:
df_ms.at[indexes_min_zero, 'minutes'] = '00'

In [30]:
df_ms.value_counts('minutes')

minutes
00    29152
25     1590
50      232
15       48
59        8
40        6
dtype: int64

In [31]:
df_ms.value_counts('seconds')

seconds
0     31028
56        4
60        2
59        2
dtype: int64

In [32]:
df_ms.seconds = df_ms.seconds.astype('string')

In [33]:
indexes_sec_zero = df_ms.seconds[df_ms.seconds == '0'].index
len(indexes_sec_zero)

31028

In [34]:
indexes_sec_zero = list(indexes_sec_zero)

In [35]:
df_ms.at[indexes_sec_zero, 'seconds'] = '00'

In [36]:
df_ms.value_counts('seconds')

seconds
00    31028
56        4
60        2
59        2
dtype: int64

In [37]:
df_ms.value_counts('hours')

hours
4    30974
5       54
3        8
dtype: int64

In [38]:
df_ms.hours = '0' + df_ms.hours.astype('string')

In [39]:
df_ms.value_counts('hours')

hours
04    30974
05       54
03        8
dtype: int64

In [40]:
p(df_ms)

(31036, 3)


,minutes,seconds,hours
0,00,00,04
1,00,00,04
2,00,00,04
3,00,00,04
4,00,00,04
31031,00,00,04
31032,00,00,04
31033,00,00,04
31034,00,00,04
31035,00,00,04


In [41]:
df_ms.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 31036 entries, 0 to 31035
Data columns (total 3 columns):
 #   Column   Non-Null Count  Dtype 
---  ------   --------------  ----- 
 0   minutes  31036 non-null  object
 1   seconds  31036 non-null  string
 2   hours    31036 non-null  string
dtypes: object(1), string(2)
memory usage: 727.5+ KB


<font color='red'> why do we have object for minutes here?????</font>

In [42]:
df_ms.minutes.value_counts()

00    29152
25     1590
50      232
15       48
59        8
40        6
Name: minutes, dtype: int64

In [43]:
df_ms.minutes = df_ms.minutes.astype('string')
df_ms.minutes

0        00
1        00
2        00
3        00
4        00
         ..
31031    00
31032    00
31033    00
31034    00
31035    00
Name: minutes, Length: 31036, dtype: string

In [44]:
df_ms['duration'] = df_ms.hours + ':' + df_ms.minutes + ':' + df_ms.seconds
p(df_ms)

(31036, 4)


,minutes,seconds,hours,duration
0,00,00,04,04:00:00
1,00,00,04,04:00:00
2,00,00,04,04:00:00
3,00,00,04,04:00:00
4,00,00,04,04:00:00
31031,00,00,04,04:00:00
31032,00,00,04,04:00:00
31033,00,00,04,04:00:00
31034,00,00,04,04:00:00
31035,00,00,04,04:00:00


In [45]:
df_ms['duration'] = pd.to_timedelta(df_ms.duration)
p(df_ms)

(31036, 4)


,minutes,seconds,hours,duration
0,00,00,04,0 days 04:00:00
1,00,00,04,0 days 04:00:00
2,00,00,04,0 days 04:00:00
3,00,00,04,0 days 04:00:00
4,00,00,04,0 days 04:00:00
31031,00,00,04,0 days 04:00:00
31032,00,00,04,0 days 04:00:00
31033,00,00,04,0 days 04:00:00
31034,00,00,04,0 days 04:00:00
31035,00,00,04,0 days 04:00:00


In [46]:
df_ms.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 31036 entries, 0 to 31035
Data columns (total 4 columns):
 #   Column    Non-Null Count  Dtype          
---  ------    --------------  -----          
 0   minutes   31036 non-null  string         
 1   seconds   31036 non-null  string         
 2   hours     31036 non-null  string         
 3   duration  31036 non-null  timedelta64[ns]
dtypes: string(3), timedelta64[ns](1)
memory usage: 970.0 KB


In [47]:
df_duration = pd.DataFrame({})

In [48]:
df_duration['duration'] = df_ms.duration
p(df_duration)

(31036, 1)


,duration
0,0 days 04:00:00
1,0 days 04:00:00
2,0 days 04:00:00
3,0 days 04:00:00
4,0 days 04:00:00
31031,0 days 04:00:00
31032,0 days 04:00:00
31033,0 days 04:00:00
31034,0 days 04:00:00
31035,0 days 04:00:00


In [49]:
df_empty = pd.concat([df_empty, df_duration], axis=1)
p(df_empty)

(31036, 30)


,GAME_ID,TEAM_ID,TEAM_NAME,TEAM_ABBREVIATION,TEAM_CITY,MIN,E_OFF_RATING,OFF_RATING,E_DEF_RATING,DEF_RATING,E_NET_RATING,NET_RATING,AST_PCT,AST_TOV,AST_RATIO,OREB_PCT,DREB_PCT,REB_PCT,E_TM_TOV_PCT,TM_TOV_PCT,EFG_PCT,TS_PCT,USG_PCT,E_USG_PCT,E_PACE,PACE,PACE_PER40,POSS,PIE,duration
0,21000961,1610612740,Hornets,NOH,New Orleans,240:00,102.5,106.9,102.1,105.7,0.3,1.1,0.382,1.44,11.5,0.319,NaN,NaN,9.916,10.3,0.462,0.512,1.0,0.198,90.42,87.0,72.50,87,0.479,0 days 04:00:00
1,21000961,1610612742,Mavericks,DAL,Dallas,240:00,102.1,105.7,102.5,106.9,-0.3,-1.1,0.710,2.00,19.0,NaN,0.681,NaN,12.211,12.6,0.493,0.554,1.0,0.191,90.42,87.0,72.50,87,0.521,0 days 04:00:00
2,21000962,1610612759,Spurs,SAS,San Antonio,240:00,125.0,124.7,110.9,118.2,14.1,6.5,0.467,1.40,18.3,0.222,0.556,0.444,16.892,16.9,0.686,0.704,1.0,0.198,91.30,88.5,73.75,89,0.610,0 days 04:00:00
3,21000962,1610612765,Pistons,DET,Detroit,240:00,110.9,118.2,125.0,124.7,-14.1,-6.5,0.524,1.83,16.2,0.444,0.778,0.556,12.793,13.6,0.484,0.511,1.0,0.197,91.30,88.5,73.75,88,0.390,0 days 04:00:00
4,21000963,1610612758,Kings,SAC,Sacramento,240:00,97.0,101.0,104.9,106.0,-7.9,-5.0,0.529,0.86,12.9,0.370,0.841,0.582,19.970,20.8,0.440,0.509,1.0,0.196,103.12,100.5,83.75,101,0.478,0 days 04:00:00
31031,21700898,1610612738,Celtics,BOS,Boston,240:00,112.5,111.2,100.9,101.0,11.6,10.2,0.641,3.13,19.1,0.231,0.830,0.515,8.258,8.2,0.523,0.557,1.0,0.200,96.98,97.5,81.25,98,0.529,0 days 04:00:00
31032,21700899,1610612741,Bulls,CHI,Chicago,240:00,85.2,87.0,102.4,104.0,-17.2,-17.0,0.515,1.21,12.7,0.297,0.840,0.535,13.709,14.0,0.387,0.422,1.0,0.198,101.82,100.0,83.33,100,0.408,0 days 04:00:00
31033,21700899,1610612751,Nets,BKN,Brooklyn,240:00,102.4,104.0,85.2,87.0,17.2,17.0,0.775,2.07,22.2,0.160,0.703,0.465,14.775,15.0,0.533,0.556,1.0,0.195,101.82,100.0,83.33,100,0.592,0 days 04:00:00
31034,21700900,1610612752,Knicks,NYK,New York,240:00,106.8,109.9,124.5,121.4,-17.7,-11.5,0.596,2.00,19.7,0.265,0.762,0.495,13.467,13.9,0.542,0.555,1.0,0.200,102.20,102.0,85.00,101,0.421,0 days 04:00:00
31035,21700900,1610612744,Warriors,GSW,Golden State,240:00,124.5,121.4,106.8,109.9,17.7,11.5,0.705,2.58,22.2,0.238,0.735,0.505,11.947,11.7,0.606,0.648,1.0,0.201,102.20,102.0,85.00,103,0.579,0 days 04:00:00


In [50]:
df_empty['duration_minutes'] = df_empty.duration.dt.seconds / 60
df_empty

,GAME_ID,TEAM_ID,TEAM_NAME,TEAM_ABBREVIATION,TEAM_CITY,MIN,E_OFF_RATING,OFF_RATING,E_DEF_RATING,DEF_RATING,E_NET_RATING,NET_RATING,AST_PCT,AST_TOV,AST_RATIO,OREB_PCT,DREB_PCT,REB_PCT,E_TM_TOV_PCT,TM_TOV_PCT,EFG_PCT,TS_PCT,USG_PCT,E_USG_PCT,E_PACE,PACE,PACE_PER40,POSS,PIE,duration,duration_minutes
0,21000961,1610612740,Hornets,NOH,New Orleans,240:00,102.5,106.9,102.1,105.7,0.3,1.1,0.382,1.44,11.5,0.319,NaN,NaN,9.916,10.3,0.462,0.512,1.0,0.198,90.42,87.0,72.50,87,0.479,0 days 04:00:00,240.0
1,21000961,1610612742,Mavericks,DAL,Dallas,240:00,102.1,105.7,102.5,106.9,-0.3,-1.1,0.710,2.00,19.0,NaN,0.681,NaN,12.211,12.6,0.493,0.554,1.0,0.191,90.42,87.0,72.50,87,0.521,0 days 04:00:00,240.0
2,21000962,1610612759,Spurs,SAS,San Antonio,240:00,125.0,124.7,110.9,118.2,14.1,6.5,0.467,1.40,18.3,0.222,0.556,0.444,16.892,16.9,0.686,0.704,1.0,0.198,91.30,88.5,73.75,89,0.610,0 days 04:00:00,240.0
3,21000962,1610612765,Pistons,DET,Detroit,240:00,110.9,118.2,125.0,124.7,-14.1,-6.5,0.524,1.83,16.2,0.444,0.778,0.556,12.793,13.6,0.484,0.511,1.0,0.197,91.30,88.5,73.75,88,0.390,0 days 04:00:00,240.0
4,21000963,1610612758,Kings,SAC,Sacramento,240:00,97.0,101.0,104.9,106.0,-7.9,-5.0,0.529,0.86,12.9,0.370,0.841,0.582,19.970,20.8,0.440,0.509,1.0,0.196,103.12,100.5,83.75,101,0.478,0 days 04:00:00,240.0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
31031,21700898,1610612738,Celtics,BOS,Boston,240:00,112.5,111.2,100.9,101.0,11.6,10.2,0.641,3.13,19.1,0.231,0.830,0.515,8.258,8.2,0.523,0.557,1.0,0.200,96.98,97.5,81.25,98,0.529,0 days 04:00:00,240.0
31032,21700899,1610612741,Bulls,CHI,Chicago,240:00,85.2,87.0,102.4,104.0,-17.2,-17.0,0.515,1.21,12.7,0.297,0.840,0.535,13.709,14.0,0.387,0.422,1.0,0.198,101.82,100.0,83.33,100,0.408,0 days 04:00:00,240.0
31033,21700899,1610612751,Nets,BKN,Brooklyn,240:00,102.4,104.0,85.2,87.0,17.2,17.0,0.775,2.07,22.2,0.160,0.703,0.465,14.775,15.0,0.533,0.556,1.0,0.195,101.82,100.0,83.33,100,0.592,0 days 04:00:00,240.0
31034,21700900,1610612752,Knicks,NYK,New York,240:00,106.8,109.9,124.5,121.4,-17.7,-11.5,0.596,2.00,19.7,0.265,0.762,0.495,13.467,13.9,0.542,0.555,1.0,0.200,102.20,102.0,85.00,101,0.421,0 days 04:00:00,240.0


In [51]:
df_empty.value_counts('duration_minutes')

duration_minutes
240.000000    29154
265.000000     1590
290.000000      232
315.000000       48
340.000000        6
239.933333        4
239.983333        2
dtype: int64

In [52]:
#LA -> Los Angeles for column 'TEAM_CITY'
df_empty.loc[(df_empty.TEAM_CITY == 'LA'), 'TEAM_CITY'] = 'Los Angeles'

In [53]:
datapath = os.path.join('..', 'processed_data')

if not os.path.exists(datapath):
    os.mkdir(datapath)

datapath_df = os.path.join(datapath, '19boxscoreadvancedv2_team_2006_2018.csv')
if not os.path.exists(datapath_df):
    df_empty.to_csv(datapath_df, index=False)